lib/users/routes.js

```diff
@@ -30,18 +35,203 @@ const all = (req, res, next) => {
+
+  _.unset(req.session, 'form.users.edit');
+  _.unset(req.session, 'form.users.save');
+
+  // something went wrong on save of existing user:
+  if (Object.keys(values).length > 0 && id) {
+    // add the previous session data back in
+    _.set(req.session, 'form.users.edit', {
+      id,
+    });
+
+    // grab the edit form and inject errors and values from req.session
+    return types.only('users', values, [merged], config).then(only => {
+      return types.form(only, errors, config).then(form => {
+        res.render('users/one', {
+          form,
+          action: `/${config.users.base}/${config.users.actions.save}`,
+          config: config.users,
+          structure: model.structure,
+          button: config.users.actions.update,
+        });
+
+        return true;
+      });
+    });
+  }
+
+  // grab the application from the database
+  return database('users').where({
+    id: req.params.id,
+  }).then(rows => {
+    // application not in database; send to 404
+    if (rows.length < 1) {
+      const err = {
+        message: config.users.messages.missing.id.replace('%id', req.params.id),
+        safe: `/${config.users.base}`,
+        status: 404,
+      };
+
+      return next(err);
+    }
+
+    const user = rows[0];
+
+    // add session data for this application
+    _.set(req.session, 'form.users.edit', {
+      id: user.id,
+    });
+
+    const data = {
+      'email': {
+        'email': {
+          'value': user.email,
+        },
+      },
+      'role': {
+        'select': {
+          'value': user.role,
+        },
+      },
+    };
+
+    return types.only('users', data, [merged], config);
+  }).then(only => {
+    return types.form(only, null, config);
+  }).then(form => {
+    res.render('users/one', {
+      form,
+      action: `/${config.users.base}/${config.users.actions.save}`,
+      config: config.users,
+      button: config.users.actions.update,
+    });
+
+
+    return true;
+  }).catch(e => {
+    const err = {
+      message: e.message,
+      safe: `/${config.users.base}`,
+      status: 404,
+    };
+
+    return next(err);
+  });
+};
 
 /**
  * Save/delete a user
+ * @param {object} req - HTTP Request
+ * @param {object} res - HTTP Response
+ *
+ * @returns {promise} database promise
  */
-const save = () => {};
+const save = (req, res) => {
+  const id = _.get(req.session, 'form.users.edit.id', null);
+  const referrer = (_.get(req.session, 'referrer') || req.get('Referrer')) || `/${config.users.base}`;
+  const merged = req.app.get('users-model') || {};
+
+  // user hit delete button
+  if (req.body.submit === config.users.actions.delete && id) {
+    return database('users').where('id', '=', id).del().then(() => {
+      _.unset(req.session, 'form.users.edit');
+      _.unset(req.session, 'form.users.save');
+
+      res.redirect(`/${config.users.base}`);
+
+      return true;
+    });
+  }
+
+  // Validation
+  const validated = types.form.validate(req.body, merged, 'save');
```

Comment:
Validation, on a new user, needs to check to ensure the email address entered isn't already in use by another user